libmavconn/src/serial.cpp

```diff
@@ -54,6 +54,15 @@ MAVConnSerial::MAVConnSerial(uint8_t system_id, uint8_t component_id,
 		serial_dev.set_option(SPB::parity(SPB::parity::none));
 		serial_dev.set_option(SPB::stop_bits(SPB::stop_bits::one));
 		serial_dev.set_option(SPB::flow_control( (hwflow) ? SPB::flow_control::hardware : SPB::flow_control::none));
+		#ifdef __linux__
+		int fd = serial_dev.native_handle();
+		struct termios tio;
+		tcgetattr(fd, &tio);
+		tio.c_cc[VTIME] = 0;
+		tio.c_cc[VMIN] = 1;
+		tcflush(fd, TCIFLUSH);
+		tcsetattr(fd, TCSANOW, &tio);
+		#endif
```

Comment:
It is possible to do that via ASIO API? Just `set_option()` looks a bit better (but it also does tty calls).

Ifdef should not have any indentation. And maybe enclosure that hack to block?
Also it definitely should have comment on what problem solves that hack.